### PR TITLE
ci: changing env vars for gh action

### LINKED ISSUE #1248: Github action: The `set-env` command is deprecated and will be disabled soon.
**Describe the bug**

The `set-env` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

image

**Describe how you solve it**

Affect all our CICD YAML

---

**Environment**

**Screenshots**

### PR DESCRIPTION
**Pull request by deepankarm**
Fixes #1248

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
       - run: |
           pip install .  --no-cache-dir
           jina check --summary-exec ./docs/chapters/all_exec.md --summary-driver ./docs/chapters/all_driver.md
-          echo ::set-env name=JINA_VERSION::"$(sed -n '/^__version__/p' ./jina/__init__.py | cut -d \' -f2)-devel"
+          echo "JINA_VERSION=$(sed -n '/^__version__/p' ./jina/__init__.py | cut -d \' -f2)-devel >> $GITHUB_ENV
```
Comment (bhavsarpratik) [inline .github/workflows/cd.yml:41]: close `"` missing?

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -74,10 +74,10 @@ jobs:
...
         run: |
           echo ${{ steps.buildx.outputs.platforms }}
-          echo ::set-env name=VCS_REF::$(git rev-parse --short HEAD)
-          echo ::set-env name=BUILD_DATE::$(date -u +"%Y-%m-%dT%H:%M:%SZ")
-          echo ::set-env name=INSTALL_DEV::"true"
-          echo ::set-env name=JINA_VERSION::"$(sed -n '/^__version__/p' ./jina/__init__.py | cut -d \' -f2)-devel"
+          echo "VCS_REF=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_ENV
+          echo "INSTALL_DEV=true" >> $GITHUB_ENV
+          echo "JINA_VERSION=$(sed -n '/^__version__/p' ./jina/__init__.py | cut -d \' -f2)-devel >> $GITHUB_ENV
```
Comment (bhavsarpratik) [inline .github/workflows/cd.yml:80]: close `"` missing?

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -25,10 +25,10 @@ jobs:
...
         run: |
           echo ${{ steps.buildx.outputs.platforms }}
-          echo ::set-env name=VCS_REF::$(git rev-parse --short HEAD)
-          echo ::set-env name=BUILD_DATE::$(date -u +"%Y-%m-%dT%H:%M:%SZ")
-          echo ::set-env name=INSTALL_DEV::"true"
-          echo ::set-env name=JINA_VERSION::"$(sed -n '/^__version__/p' ./jina/__init__.py | cut -d \' -f2)-devel"
+          echo "VCS_REF=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_ENV
+          echo "INSTALL_DEV=true" >> $GITHUB_ENV
+          echo "JINA_VERSION=$(sed -n '/^__version__/p' ./jina/__init__.py | cut -d \' -f2)-devel >> $GITHUB_ENV
```
Comment (bhavsarpratik) [inline .github/workflows/nightly.yml:31]: close `"` missing?

**Review by bhavsarpratik — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (deepankarm) [reply .github/workflows/cd.yml]: Good catch, fixed.

**Review by bhavsarpratik — APPROVED** [review]

**Review by hanxiao — APPROVED** [review]
i've tested all lines in local bask, works.

Ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable

**Pull request merged**